### PR TITLE
Adds alerting rule crd creation logic at startup of the operator

### PR DESCRIPTION
**Pull request by prateeknayak**
- Adds[ creation of alerting rule crd](https://github.com/MYOB-Technology/ops-kube-alerting-rules-operator/compare/feature/create-crd-on-boot?expand=1#diff-d1253354b691edb0a165e202b86a4951) if not present
- Updates most of the auto-generated code in pkg due to change in dependencies.
- Fixes linting issues found by `gometalinter`
- Updates Dockerfile to use `myobplatform/go-dep:1.9` image which includes standard go tools.

### CODE REVIEW

File: pkg/crd/alertingRules_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestCreateAlertingRuleCRD(t *testing.T) {
+	c := fake.NewSimpleClientset()
+	c.ClearActions()
+
+	for i := 1; i <= 5; i++ {
+		// create CRD multiple times it should not barf
+		scenario(t, c, i)
```
Comment (pvdvreede) [inline pkg/crd/alertingRules_test.go]: like the use of this helper function, good name to.

File: pkg/crd/alertingRules.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+	"github.com/MYOB-Technology/ops-kube-alerting-rules-operator/pkg/apis/alertingrule/v1alpha1"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Create Alerting Rule CRD
+// If it already exists, it should return and Already Exists Error
```
Comment (pvdvreede) [inline pkg/crd/alertingRules.go]: I think this works the way I would expect but the comment is a bit vague, I assume that it creates the CRD and is a noop if it already exists, ie it does not return the error if it already exists...

File: Makefile [hunk 3]
```diff
@@ -1,21 +1,24 @@
 .PHONY: publish publish-version login fmt ci
 
 GOFILES_NOVENDOR = $(shell find . -type f -name '*.go' -not -path "./vendor/*")
+GO_TEST_PKGS = $(shell go list ./... | grep -v github.com/MYOB-Technology/ops-kube-alerting-rules-operator/pkg/client | grep -v github.com/MYOB-Technology/ops-kube-alerting-rules-operator/pkg/signals)
```
Comment (pvdvreede) [inline Makefile]: Good idea, hopefully this improves the compile speed for tests...

Is there still an apis package as well that is autogenerated and has no tests?

File: Makefile [hunk 4]
```diff
@@ -1,21 +1,24 @@
...
 REPO := myobplatform/alerting-rules-operator
 
 define docker_push_image
 	docker build --build-arg VERSION=${TAG} -t ${REPO}:${TAG} .
 	docker push ${REPO}:${TAG}
 endef
 
+dummy:
```
Comment (pvdvreede) [inline Makefile]: Do we still need this dummy target?

File: pkg/crd/alertingRules.go [hunk 5]
```diff
@@ -0,0 +1,35 @@
+package crd
```
Comment (pvdvreede) [inline pkg/crd/alertingRules.go]: Im wondering whether this file could just be in the controller package? its sole use is with the controller, so Im not sure its worth having a lot of single file small packages...

**Comment by pvdvreede** [comment]
Oh just a general question, what is the go convention for file names? I have been using snake case for the file name, but camel case for variable names... Good to know the convention so either you or me can change to be consistent as we are naming our files with different conventions...

File: pkg/controller/alerting_rules_crd_test.go [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+	c := fake.NewSimpleClientset()
+	c.ClearActions()
+
+	for i := 1; i <= 5; i++ {
+		// create CRD multiple times it should not barf
+		scenario(t, c, i)
+	}
+}
+
+func TestCreateAlertingRuleCRD2(t *testing.T) {
```
Comment (pjmichael) [inline pkg/controller/alerting_rules_crd_test.go]: Could we perhaps rename this (or the one before Eg "TestCreateAlertingRuleIsIdempotent"?).

**Pull request merged**